docs/sources/flow/concepts/clustering.md

```diff
@@ -0,0 +1,303 @@
+---
+title: Clustering
+weight: 500
+labels:
+  stage: beta
+---
+
+# Clustering (beta)
+
+Clustering enables Grafana Agent Flow to coordinate a fleet of agents to
+work together for workload distribution and high availability. It makes agent
+deployments horizontally scalable by default, and minimizes the operational
+overhead of managing your agent infrastructure to handle demand.
+
+To achieve this, Grafana Agent makes use of an eventually consistent model and
+the lightweight [grafana/ckit][] framework. All running nodes are assumed to
+use the same configuration and have access to the same type of hardware and
+network resources such as PVCs and service discovery APIs.
+
+Clustering is built from the ground-up to be the de-facto answer to "how do I
+scale my Grafana Agent setup" in the future. As an example, in comparison to a
+setup using [hashmod sharding][], clustering with target auto-distribution
+scales and provides resiliency at _half_ the cost in resources and provides
+dynamic resharding without the need to change configuration files.
+
+The behavior of a standalone, non-clustered agent is the same as if it was a
+1-node cluster.
+
+## Usage
+
+The following set of command-line flags are used to configure clustering:
```

Comment:
This feels like detailed reference documentation to me, plus we already document these flags in the cli page. Should we link to that page here?

Reply:
I removed the documentation from the `run` command page when I tried to gather everything here.

Since you think it reads too heavy, let's separate the info back and see how it looks like!